controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -575,6 +594,23 @@ func validateControlPlaneSpec(rosaClient *rosa.RosaClient, rosaScope *scope.ROSA
 		return &message, nil
 	}
 
+	// validate Etcd Encryption KMS Arn
+	err = kmsArnRegexpValidator.ValidateKMSKeyARN(&rosaScope.ControlPlane.Spec.EtcdEncryptionKMSArn)
+	if err != nil {
+		return nil, err
```

Comment:
the function should return a failure message if something is invalid. returning an error means there was a real error during the validation process, like a failed API call.

Reply:
this validation can also be moved to the webhook similar to tags.

Reply:
The validation function has been moved to the webhook as suggested.